client/src/views/workspaces/WorkspaceSharingModal.vue

```diff
@@ -310,6 +478,68 @@ async function updateUserRole(
   }
   await refreshSharingInfo();
 }
+
+async function onBatchRoleChange(newRoleOption: MsOption): Promise<void> {
+  let errorCount: number = 0;
+  let error = undefined;
+
+  for (const userRole of selectedUsers.value) {
+    if (!userRole.role || newRoleOption.key !== userRole.role) {
+      const result = await shareWorkspace(props.workspaceId, userRole.user.id, newRoleOption.key);
+      if (!result.ok) {
+        if (errorCount === 0) {
+          error = result.error;
+        }
+        errorCount += 1;
+      }
+    }
+  }
+
+  if (errorCount > 0 && errorCount < selectedUsers.value.length) {
```

Comment:
Technically no, since you only call `shareWorkspace` on some of the selected users (condition on line 487), you could have a case where all operations failed with a number of errors inferior to the number of selected users. You do need to know how many attempts were made.